src/caip-types.ts

```diff
@@ -13,11 +13,17 @@ export const CAIP_ACCOUNT_ID_REGEX =
 
 export const CAIP_ACCOUNT_ADDRESS_REGEX = /^[-.%a-zA-Z0-9]{1,128}$/u;
 
+export const CAIP_ASSET_NAMESPACE_REGEX = /^[-a-z0-9]{3,8}$/u;
+
+export const CAIP_ASSET_REFERENCE_REGEX = /^[-.%a-zA-Z0-9]{1,128}$/u;
+
+export const CAIP_TOKEN_ID_REGEX = /^[-.%a-zA-Z0-9]{1,78}$/u;
+
 export const CAIP_ASSET_TYPE_REGEX =
   /^(?<chainId>(?<namespace>[-a-z0-9]{3,8}):(?<reference>[-_a-zA-Z0-9]{1,32}))\/(?<assetNamespace>[-a-z0-9]{3,8}):(?<assetReference>[-.%a-zA-Z0-9]{1,128})$/u;
 
 export const CAIP_ASSET_ID_REGEX =
-  /^(?<chainId>(?<namespace>[-a-z0-9]{3,8}):(?<reference>[-_a-zA-Z0-9]{1,32}))\/(?<assetNamespace>[-a-z0-9]{3,8}):(?<assetReference>[-.%a-zA-Z0-9]{1,128})\/(?<tokenId>[-.%a-zA-Z0-9]{1,78})$/u;
+  /^(?<chainId>(?<namespace>[-a-z0-9]{3,8}):(?<reference>[-_a-zA-Z0-9]{1,32}))\/(?<assetNamespace>[-a-z0-9]{3,8}):(?<assetReference>[-.%a-zA-Z0-9]{1,128})(\/(?<tokenId>[-.%a-zA-Z0-9]{1,78}))?$/u;
```

Comment:
The `token_id` is not optional for an [`asset_id`](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-19.md#syntax-1).

Reply:
Would you mind double-checking the spec again? It discusses it as being optional (as I mentioned above), and it gives many examples that demonstrate is as optional.

Reply:
Actually, now I've re-read the spec for the millionth time, and I can see that it was written in a way that can be  interpreted in two different ways. I see now that your interpretation may be more valid than mine. I'll change this PR to reflect it.